libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -313,7 +333,11 @@ void AP_OSD_Screen::draw_speed_vector(uint8_t x, uint8_t y,Vector2f v, int32_t y
         int32_t interval = 36000 / SYM_ARROW_COUNT;
         arrow = SYM_ARROW_START + ((angle + interval / 2) / interval) % SYM_ARROW_COUNT;
     }
-    backend->write(x, y, false, "%c%3.0f%c", arrow,  v_length * 3.6, SYM_KMH);
+    if (osd->options.get() & AP_OSD::OPTION_IMPERIAL_UNITS) {
```

Comment:
I think we need a helper function for writing with units, something like:
  write_units(char prefix_char, const char *fmt, float value);

Reply:
Added function to convert values based on chosen units system.